utest/ui/DataFilters.cpp

```diff
@@ -442,6 +440,94 @@ TEST_F(DataFilterTest, MaxPointCountDataPointsFilter)
 	validate3dTransformation();
 }
 
+TEST_F(DataFilterTest, OctreeGridDataPointsFilter)
+{
+	const unsigned int nbPts = 60000;
+	const DP cloud = generateRandomDataPoints(nbPts);	
+	params = PM::Parameters(); 
+	
+//CASE (1): 1/pts by octants + validate parallel build
+// the number of point should not change
+// the sampling methods should not change anything
+	PM::DataPointsFilter* octreeFilter;
+	
+	const vector<int> samplingMethods = list_of (0) (1) (2);
+	const vector<bool> parallels = list_of (true) (false);
+	
+	for(const auto& par : parallels)
+	{
+		for(const auto& meth : samplingMethods)
+		{
+			params.clear();
+			params["buildMethod"] = "0";
+			params["maxPointByNode"] = "1";
+			params["samplingMethod"] = toParam(meth);
+			params["buildParallel"] = toParam(par);
+		
+			octreeFilter = 
+				PM::get().DataPointsFilterRegistrar.create("OctreeGridDataPointsFilter", params);
+	
+			const DP filteredCloud = octreeFilter->filter(cloud);
+		
+			//Check number of points
+			EXPECT_EQ(cloud.getNbPoints(), filteredCloud.getNbPoints());
+			EXPECT_EQ(cloud.getDescriptorDim(), filteredCloud.getDescriptorDim());
+			EXPECT_EQ(cloud.getTimeDim(), filteredCloud.getTimeDim());
+		
+			EXPECT_EQ(filteredCloud.getNbPoints(), nbPts);
+		}
+	}
+	
+//CASE (2): 200/pts by octants	
+	for(const auto& meth : samplingMethods)
```

Comment:
Couldn't you just make another loop around the nested loop above over "1", "200" for maxPointByNode ? What else is specific here? That way you would test more combinations, which is probably good.